PWGHF/TableProducer/treeCreatorB0ToDPi.cxx

```diff
@@ -0,0 +1,384 @@
+                  full::Pt,
+                  full::Eta,
+                  full::Phi,
+                  full::Y,
+                  hf_cand_3prong::FlagMcMatchRec,
+                  hf_cand_3prong::OriginMcGen);
+} // namespace o2::aod
+
+/// Writes the full information in an output TTree
+struct HfTreeCreatorB0ToDPi {
+  Produces<o2::aod::HfCandB0Full> rowCandidateFull;
+  Produces<o2::aod::HfCandB0FullEvents> rowCandidateFullEvents;
+  Produces<o2::aod::HfCandB0FullParticles> rowCandidateFullParticles;
+  Produces<o2::aod::HfCandB0Lite> rowCandidateLite;
+
+  Configurable<int> selectionFlagB0{"selectionB0", 1, "Selection Flag for B0"};
+  Configurable<bool> fillCandidateLiteTable{"fillCandidateLiteTable", false, "Switch to fill lite table with candidate properties"};
+
+  // parameters for production of training samples
+  Configurable<bool> fillOnlySignal{"fillOnlySignal", false, "Flag to fill derived tables with signal for ML trainings"};
+  Configurable<bool> fillOnlyBackground{"fillOnlyBackground", false, "Flag to fill derived tables with background for ML trainings"};
+  Configurable<float> downSampleBkgFactor{"downSampleBkgFactor", 1., "Fraction of background candidates to keep for ML trainings"};
+  Configurable<float> ptMaxForDownSample{"ptMaxForDownSample", 10., "Maximum pt for the application of the downsampling factor"};
+
+  Filter filterSelectCandidates = aod::hf_sel_candidate_b0::isSelB0ToDPi >= selectionFlagB0;
+  using SelectedCandidatesMc = soa::Filtered<soa::Join<aod::HfCandB0, aod::HfCandB0McRec, aod::HfSelB0ToDPi>>;
+
+  Partition<SelectedCandidatesMc> recSig = nabs(aod::hf_cand_b0::flagMcMatchRec) == (int8_t)BIT(aod::hf_cand_b0::DecayTypeMc::B0ToDplusPiToPiKPiPi);
+  Partition<SelectedCandidatesMc> recBg = nabs(aod::hf_cand_b0::flagMcMatchRec) != (int8_t)BIT(aod::hf_cand_b0::DecayTypeMc::B0ToDplusPiToPiKPiPi);
+
+  void init(InitContext const&)
+  {
+  }
+
+  template <typename T>
+  void fillEvent(const T& collision, int isEventReject, int runNumber)
+  {
+    rowCandidateFullEvents(
+      collision.bcId(),
+      collision.numContrib(),
+      collision.posX(),
+      collision.posY(),
+      collision.posZ(),
+      isEventReject,
+      runNumber);
+  }
+
+  template <bool doMc = false, typename T, typename U>
+  void fillCandidateTable(const T& candidate, const U& prong1)
+  {
+    int8_t flagMc = 0;
+    int8_t originMc = 0;
+    if constexpr (doMc) {
+      flagMc = candidate.flagMcMatchRec();
+      originMc = candidate.originMcRec();
+    }
+    if (fillCandidateLiteTable) {
+      rowCandidateLite(
+        candidate.chi2PCA(),
+        candidate.decayLength(),
+        candidate.decayLengthXY(),
+        candidate.decayLengthNormalised(),
+        candidate.decayLengthXYNormalised(),
+        candidate.ptProng0(),
+        candidate.ptProng1(),
+        candidate.impactParameter0(),
+        candidate.impactParameter1(),
+        candidate.impactParameter0() * candidate.impactParameter1(),
```

Comment:
@AlexBigO @fgrosa , how is it useful to save both impact parameters and also their product?

Reply:
Indeed, we can spare disk space and compute it offline if needed. Sorry for the previous comment!